pkg/process/config/yaml_config.go

```diff
@@ -172,6 +172,23 @@ func (a *AgentConfig) loadSysProbeYamlConfig(path string) error {
 		a.Windows.DriverBufferSize = driverBufferSize
 	}
 
+	// If there is no network_config, assume it is an old version of the config, and enable
+	// the network check. This has the downside that if the network_config is deleted, network
+	// check will be enabled.  However, this seems like the best option for backwards compatibility.
+	//
+	// The network check won't be run if sysprobe isn't enabled so there's no need to check
+	// EnableSystemProbe here.
+	networkEnabled := true
+	if config.Datadog.IsSet(key(spNS, "network_config")) {
```

Comment:
why not set by default "enabled=true"

Reply:
Originally the logic was a bit more complicated, but after simplifying it, we wanted to message like we are doing here (emit value if found, emit message about default if not found).  You're correct though, the logic for setting the value has become a slightly twisty parody of "default enabled" :)  I can fix this a bit to make it a bit cleaner.

Reply:
System-probe handles defaults and config ... differently.  Sounds like it will be refactored in an upcoming ticket (for security module work).  What I added mimics how TCP Queue Length and OOM Kill modules are enabled/disabled (via EnabledChecks - another existing hack).